src/mochiweb_socket_server.erl

```diff
@@ -273,8 +271,11 @@ handle_call({get, Property}, _From, State) ->
     {reply, Res, State};
 handle_call(stop, _From, State) ->
     {stop, normal, ok, State};
-handle_call(prep_stop, _From, State) ->
-    {reply, close_listen_socket(State), State};
+handle_call(prep_stop, From, State) ->
+    close_listen_socket(State),
+    State1 = State#mochiweb_socket_server{shutdown_notify_pid=From, acceptor_pool_size=0},
+    % Reply will be given when active_socket count goes to 0
+    {noreply, State1};
```

Comment:
Removed the polling logic by using a noreply response and a reply when active_sockets become 0.